### PR TITLE
feat: compliance rules highlight activities on popover click

### LINKED ISSUE #62: [FEAT] Compliances Rules: highlight the activities involved in the violations
**Is your feature request related to a problem? Please describe.**
The popover currently displays the elements with violations but we don't see them on the diagram.

**Describe the solution you'd like**
When hovering on an element of the "compliance issues" table, highlight the related activity.

For instance, in the following, when the mouse hover the "Clear Invoice", the "Record Goods Receipt" activity is highlighted in yellow
image

**Additional context**
Be aware of #4 and #58 prior implementing this request

### PR DESCRIPTION
**Pull request by assynour**
closes #62 

icpm-popover

### CODE REVIEW

File: src/compliance-rules.js [hunk 1]
```diff
@@ -233,44 +260,25 @@ function registerBpmnElement(bpmnElement) {
...
+        <th>%traces</th>
+      </tr>`
+  
+  for (let key in activityComplianceData) {
+    popoverData += `<tr>`
+    popoverData += `<td>`+ key +`</td>`
+    popoverData += `<td>`+ activityComplianceData[key]["nbViolations"] +`</td>`
+    popoverData += `<td>`+ activityComplianceData[key]["percentTraces"] +`</td>`
+    popoverData += `</tr>`
   }
```
[the remark this answers is not in the document]
Comment (assynour) [reply src/compliance-rules.js]: fixed in 98a566ba7633ae98f4be176dbcf1404572e34be3

File: src/compliance-rules.js [hunk 2]
```diff
@@ -218,6 +223,37 @@ function addPopover(activityId, bpmnVisualization) {
...
+      const activityName = selectedRow.firstElementChild.textContent;
+      const activityId = getElementIdByName(activityName)
+      //highlight activity
+      bpmnVisualization.bpmnElementsRegistry.removeCssClasses(activityId, "cause-violation")
+      // workaround for https://github.com/process-analytics/icpm-demo-2022/issues/87
+      if(complianceData.has(activityName)){
+        addRippleCircles(activityId, bpmnVisualization)
+      }
+    }
+  });
```
Comment (tbouffard) [inline src/compliance-rules.js:255]: **issue:** This should be done with CSS directly, as this is done in https://github.com/process-analytics/bonita-day-demo-2023/pull/10/files

**Review by tbouffard — CHANGES_REQUESTED** [review]
Design and refactoring to plan

File: src/compliance-rules.js [hunk 3]
```diff
@@ -218,6 +223,37 @@ function addPopover(activityId, bpmnVisualization) {
     trigger: "click",
   });
 
+  // Add mouseover and mouseout event listeners to the table rows
+  tippyInstance.popper.addEventListener("mouseover", (event) => {
+    if (event.target.nodeName === "TD") {
+      const selectedRow = event.target.parentElement;
```
Comment (tbouffard) [inline src/compliance-rules.js:229]: **suggestion:** https://github.com/process-analytics/bonita-day-demo-2023/pull/18/files registers the event listener directly in the elements of the HTML of the popover. This is easy to register them directly on the rows (TR elements).

File: src/compliance-rules.js (hunk 2 above)
Comment (tbouffard) [reply src/compliance-rules.js:255]: @assynour did you have time to check the review?

Comment (assynour) [reply src/compliance-rules.js:255]: I will fix it this week

File: src/compliance-rules.js (hunk 3 above)
Comment (assynour) [reply src/compliance-rules.js:229]: I will take this into account

File: src/compliance-rules.js (hunk 2 above)
Comment (assynour) [reply src/compliance-rules.js]: fixed in 57b6053372edb1c7c3baf5202925c300819e9f32

File: src/compliance-rules.js (hunk 3 above)
Comment (assynour) [reply src/compliance-rules.js]: fixed in 6e0c3006abeffc892f1c31bd58aa63c1bb32e551

File: src/compliance-rules.js [hunk 4]
```diff
@@ -216,11 +221,75 @@ function addPopover(activityId, bpmnVisualization) {
...
+  manageEventListeners(instance, bpmnVisualization, true);
+}
+
+function unregisterEventListeners(instance, bpmnVisualization){
+  manageEventListeners(instance,bpmnVisualization,false);
+}
+
+function manageEventListeners(instance, bpmnVisualization, register){
+  // Target instance.popper. Keep using document for now as the previous popper (when going back to the subprocess after a first venue)
+  // may still exist in the DOM of the subprocess view
```
Comment (tbouffard) [inline src/compliance-rules.js:245]:
```suggestion
```

**Review by tbouffard — APPROVED** [review]
LGTM, I didn't check the code in details, but it does the job and reuse what has been done in the bonita-day demo
✔️ Tested with the surge preview environment


https://github.com/process-analytics/icpm-demo-2022/assets/27200110/1e29ef21-29b9-4a76-a66b-fb35b5c15480

**Pull request merged**